packages/ui/src/components/va-virtual-scroller/VaVirtualScroller.stories.ts

```diff
@@ -1,12 +1,285 @@
+  })
+
+  await step('disabled scroller loads all items', async () => {
+    expect(list.children.length).toBe(1000)
+  })
+}
+
+export const Horizontal = () => ({
+  components: { VaVirtualScroller },
+  data: () => ({ hugeArray }),
+  template: `
+    <va-virtual-scroller
+      :items="hugeArray"
+      horizontal
+      :bench="10"
+      :wrapper-size="200"
+      v-slot="{item, index}"
+    >
+      <div class="whitespace-nowrap">{{ index }} - {{ item }} |&nbsp;</div>
+    </va-virtual-scroller>
+  `,
+})
+
+Horizontal.play = async ({ step }) => {
+  const scroller = document.querySelector('.va-virtual-scroller') as HTMLElement
+
+  await step('the width of wrapper of horizontal scroller is set to wrapper-size', async () => {
+    expect(scroller.style.width).toEqual('200px')
+  })
+}
+
+export const AutoWrapperSize = () => ({
+  components: { VaVirtualScroller },
+  data: () => ({ hugeArray }),
+  template: `
+    <div class="container" style="height: 200px;">
+      <va-virtual-scroller
+        :items="hugeArray"
+        wrapper-size="auto"
+        v-slot="{item}"
+      >
+        <div>{{ item }}</div>
+      </va-virtual-scroller>
+    </div>
+  `,
+})
+
+AutoWrapperSize.play = async ({ step }) => {
+  const scroller = document.querySelector('.va-virtual-scroller') as HTMLElement
+  const container = document.querySelector('.container') as HTMLElement
+
+  await step('wrapper size equals the size of the container div', async () => {
+    expect(scroller.style.height).toEqual(container.style.height)
+  })
+}
+
+export const Bench = () => ({
+  components: { VaVirtualScroller },
+  data: () => ({ hugeArray }),
+  template: `
+    <p>[bench = 0]</p>
+    <va-virtual-scroller
+      :items="hugeArray"
+      :bench="0"
+      :wrapper-size="200"
+      v-slot="{item, index}"
+    >
+      <div>{{ index }} - {{ item }}</div>
+    </va-virtual-scroller>
+    <p>[bench = 20]</p>
+    <va-virtual-scroller
+      :items="hugeArray"
+      :bench="20"
+      :wrapper-size="200"
+      v-slot="{item, index}"
+    >
+      <div>{{ index }} - {{ item }}</div>
+    </va-virtual-scroller>
+  `,
+})
+
+Bench.play = async ({ canvasElement, step }) => {
+  const scrollers = document.querySelectorAll('.va-virtual-scroller')
+  const canvas = within(canvasElement)
+  const lists = canvas.getAllByRole('list')
+
+  await step('wrapper height is 200px', async () => {
+    expect((scrollers[0] as HTMLElement).style.height).toEqual('200px')
+    expect((scrollers[1] as HTMLElement).style.height).toEqual('200px')
+  })
+
+  await step('the virtual scroller that has 20 bench is 40 items larger than no bench virual scroller', async () => {
+    expect(lists[1].children.length - lists[0].children.length).toBe(40)
+  })
+}
+
+export const CustomKey = () => ({
+  components: { VaVirtualScroller },
+  data: () => ({ hugeObjectsArray }),
+  template: `
+    <va-virtual-scroller
+      :items="hugeObjectsArray"
+      :bench="10"
+      track-by="value"
```

Comment:
You asked about how to test this story. I think we can ignore this for now.

I moved your concern to #3865, which is like 10 times less prio than current effort.